python/samples/demos/guided_conversations/guided_conversation/utils/services.py

```diff
@@ -0,0 +1,33 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+from dotenv import dotenv_values
+from semantic_kernel import Kernel
+from semantic_kernel.connectors.ai.open_ai import AzureChatCompletion, OpenAIChatCompletion
+
+
+def add_service(kernel: Kernel, service_id: str) -> Kernel:
+    """Configure the AI service for the kernel
+
+    Args:
+        kernel (Kernel): The kernel to configure
+
+    Returns:
+        Kernel: The configured kernel
+    """
+    config = dotenv_values(".env")
```

Comment:
We use Pydantic settings to leverage env vars, if available, otherwise, one can point to a `.env` file, or even override with an api_key as needed (see [here](https://github.com/microsoft/semantic-kernel/blob/c15e253097c6805752cdc62ade4e05ccce290e3a/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion.py#L52C13-L58C14) as an example). I think it'd be great to align with that here.

Reply:
Is there an example in samples that shows how to configure a AzureChatCompletion service that uses RBAC? Seems like with SK>=1.2.0 `AzureChatCompletion(service_id=service_id)` no longer works, but the samples I'm coming across reference that usage.

Reply:
You'd need to explicitly pass in a token provider to the class' constructor. We don't have an explicit example around this, but you can see that the constructor accepts this and will use this if provided:

https://github.com/microsoft/semantic-kernel/blob/68b21b81df441fabfdebdb6c5867265308175950/python/semantic_kernel/connectors/ai/open_ai/services/azure_chat_completion.py#L50

Reply:
`AZURE_OPENAI_CHAT_DEPLOYMENT_NAME` is still a required parameter. If not provided via env vars or the .env file, it will need to be passed in via the constructor as the `deployment_name`.

Reply:
I removed this file and instead add the AzureChatCompletion service directly in the script or notebooks.
I figure it should be fairly obvious that this is where someone using this should configure their own service. But let me know if I should abstract this in another way (downside of this approach is there's duplication across the notebooks).

Reply:
Hi @DavidKoleczek, in just the last week or so, we introduced a default token provider as part of the AzureChatCompletion class: 

https://github.com/microsoft/semantic-kernel/blob/fd6d4a17569e56257ba9df0d5b17e28ab53cb88e/python/semantic_kernel/connectors/ai/open_ai/services/azure_chat_completion.py#L106C1-L108C14

This will get a token via DefaultAzureCredential, if the api_key, ad_token, ad_token_provider or the async_client are None.

May be worth updating your code to remove the need to manually fetch a token.

Reply:
Done! 
FYI I had to set the SK dependency to the main git branch, rather than using the latest version on PyPI since I think that change is not available in v1.9. This can be changed in the `pyproject.toml` by changing `semantic-kernel = { git = "https://github.com/microsoft/semantic-kernel.git", branch = "main", subdirectory = "python" }` to `semantic-kernel = "^1.10"` or whatever the version ends up being.